### ISSUE TITLE
Show reflog

### ISSUE THREAD

**Issue by t-gehring** [mhutchie/vscode-git-graph#89]
**Describe the feature that you'd like**
Add an option to display reflog commits, like when running `git log --reflog`

**Additional context (optional)**
Ideally this would be a checkbox in the UI or something, rather than a setting, as it's the sort of thing you'd turn on temporarily rather than all of the time. It would also be nice if they're visually distinct from normal commits, maybe greyed out?

**Comment by mhutchie** [comment]
Thanks for this suggestion! This functionality would definitely be great to have in Git Graph. I completely agree with your thoughts on how it should be incorporated!

**Comment by mhutchie** [comment]
This issue has been left open for 7 months for others to express interest in this feature. As only one other Git Graph user has indicated they would like this functionality in this period, it indicates to me that this feature would be used by very few users.

Although this functionality would definitely be nice to have, and be useful under certain circumstances, the lack of user interest is indicating that it will deliver little overall value for the current Git Graph user community. Because of this, and that it is significantly complicated to appropriately integrate in the UI, I can't justify the time required to implement it at the present time. I'll close this feature request for now and move it to the backlog.

Note to other users reading this in the future: If you'd like this functionality, please reply to this thread to express your interest. I'd happily re-open and add this feature if there was any indication it would be used by a decent number of users.

**Comment by yokljo** [comment]
I frequently run into situations where I would like to have this feature. If I accidentally make a commit off a branch then switch branches, the new commit disappears. It seems I can only fix this situation from a terminal.

**Comment by t-gehring** [comment]
I recently took another look at this, as it's still a feature which I think would be incredibly valuable. When I first tried to hack it in myself last year, I quickly discovered that it wasn't something the back-end would support as you pointed out @mhutchie .

However, doing a little experiment just now, I've found that due to the various back-end improvements that have made to the graph since, reflog works perfectly just by adding '--reflog' into the args variable of DataSource::getLogs()! Enabling the feature properly would just be a matter of hooking up UI to enable that.

I'd offer to create a pull request to add it in myself, but I doubt I'll have the spare time any time soon. But I'd be very grateful if someone else manages to do that, or if @mhutchie can find the time to fit it in.

**Comment by t-gehring** [comment]
In fact, I've created [a patch to enable reflog](https://github.com/mhutchie/vscode-git-graph/files/4250372/git-graph-reflog.patch.txt). I'm not used to the github workflow, so if someone else wants to create a PR with that code, it'll probably get stuff happening faster than waiting for me to figure out how to do that.

**Comment by mhutchie** [comment]
Thanks @t-gehring! I’ll have a look at your patch after the next release is out (hopefully this weekend), as the release will already be very large with loads of new features.

**Comment by mhutchie** [comment]
I'm re-opening this feature request, and intend to include it in the next available release of Git Graph, as:
* There has been an increased rate of user interest for it.
* I've read up & learnt more about the reflog, and there doesn't appear to be any undesirable side-effects of including `--reflog` in the `git log` command (which is used get the commits for the Git Graph View).
    * Since this feature request was first raised, other improvements to Git Graph have resolved all of the issues I found when first investigating the feasibility of this feature request last year.
* @t-gehring has kindly provided a working proof of concept (which I've had a look at, and tried out now that Git Graph 1.22.0 was released last weekend).

**Comment by mhutchie** [comment]
This will be available in v1.23.0.

If you'd like to use it before the next release, you can download [v1.23.0-beta.2](https://github.com/mhutchie/vscode-git-graph/releases/tag/v1.23.0-beta.2), and install it following the instructions provided [here](https://github.com/mhutchie/vscode-git-graph/wiki/Downloading-&-Installing-Releases#manual-installation).

**Comment by t-gehring** [comment]
I've downloaded the latest beta to check this out, seems to be working perfectly. Fwiw I think that where you've put it in the UI is a lot better than in my proof of concept. Thank you so much for including this!

**Comment by dsvost** [comment]
It can be only for me, but this checkbox description a bit confusing (which i think was added by this effort). I not native English and for me this sounds like "Show in graph only commits which also exist in reflog.". As I said, this is can be only for me it is that way :). Also, I found that only this option make stashes appear at correct places, otherwise stash commits sometimes appears on wrong commits visually in graph. Not sure if such issue appeared somewhere before. I mean, for me it works only if I disable Show stashes checkbox completely and use only this one to show commits from reflog as well (where all stashes are as well). And even I not see clear sign/mark that particular commit in graph is a stash, It is still fine, cause easy to figure out, case all my stashes has "WIP" prefix.
![image](https://github.com/user-attachments/assets/28336abf-b8a2-430f-8a72-ec604be5e319)
